packages/shared-lib/src/lib/lib.ts

```diff
@@ -1,2 +1,12 @@
 export type Time = number
 export type TimeDuration = number
+
+export function assertNever(_never: never): void {
```

Comment:
I'm curious. Is this method usefull?
If you try to call it, wouldn't you just get an error saying you are trying to pass something to type never, making a call to this method uncompilable?
If above is true - then what's the purpose of having a method in the code we can't call?
Or am I missing something about the "never" type?

Reply:
This method is for doing some type checks. You can't pass any valid value to this method (number/string/object/any is not assignable to never). So we use this method to help us make sure that a path is dead code.

For example, if a new value `Test = 'test'` was added to the enum, the typescript compiler would error on the line this was called saying `Test is not assignable to never`. 
Without using `assertNever` we would be relying on manual testing and unit testing to ensure that everywhere we branch based on the enum value handles the new value. But instead we can use the typescript compiler error as a much quicker view of places that need to be updated. It doesnt catch everything, but it will catch a lot of them

Reply:
The reason I am asking is because I tried using it once and if I parsed something as a parameter I got a compiler error saying that I couldn't parse something to type never.
If I didn't parse any arguments I got a compiler error saying that the method was expecting a parameter but got none.
Either way, I couldn't compile my code if I tried to use it, so I fail to see the use of it.
But maybe I am just using it wrong?

Reply:
@LindvedKrvang you can see it in use here: https://github.com/nrkno/sofie-core/blob/a6fe9c87c3ce2a2e9ca627368a77bd47fcf95e7b/packages/job-worker/src/playout/timeline/piece.ts#L52 and here: https://github.com/nrkno/sofie-core/blob/a12f7d8082fe1c08b5c68d6d536cbc5b32c1aebf/meteor/lib/api/triggers/actionFactory.ts#L545

I find `assertNever` insanely useful. Makes adding new features very easy, as long as you're following a certain pattern of typing objects. If you passed something as a parameter, and got a compiler error that means that whatever you were passing was in fact not of type `never`, which means that there was some case that you haven't handled in your code and it was in fact possible to get into the codepath where `assertNever` was used: either because of not-so-strict typing or an actual case that was unhandled. The good thing is that the error that TypeScript will give on that `assertNever` will hint at what case needs to be handled.

Reply:
Here's an example:
```
enum Animal {
	DOG,
	CAT,
}
function isRound(animal: Animal): boolean {
	switch (animal) {
		case Animal.DOG:
			return true
		case Animal.CAT:
			return false
		default:
			// Ensure we've handled all of the cases:
			assertNever(animal)
			return false
	}
}
```
This compiles without errors.
But if you add a `FISH` to the enum, you'll get a compiler error as a reminder that you have to remember to handle that case in the switch too. So it's basically a guard against future bugs/unhandled cases.